src/StackExchange.Redis.MultiplexerPool/Multiplexers/InternalDisposableConnectionMultiplexer.cs

```diff
@@ -22,6 +23,10 @@ public InternalDisposableConnectionMultiplexer(IConnectionMultiplexer wrappedCon
         /// <inheritdoc />
         public void Dispose()
             => throw CreateDisposeNotAllowedException();
+        
+        /// <inheritdoc />
+        public ValueTask DisposeAsync()

```

Comment:
I am pretty sure that this will cause unobserved task exception, can you add a UT and check it ?